lib/ansible/modules/network/asa/asa_og.py

```diff
@@ -0,0 +1,804 @@
+#!/usr/bin/python
+#
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: asa_og
+version_added: "2.8"
+author:
+    - "Federico Olivieri (@Federico87)"
+short_description: Manage object groups on a Cisco ASA
+description:
+    - This module allows you to create and update object-group network/service on Cisco ASA device.
+extends_documentation_fragment: asa
+options:
+    name:
+        description:
+            - object-group name
+        required: true
+    group_type:
+        description:
+            - object-group type
+        choices: ['network-object', 'service-object', 'port-object']
+        required: true
+    protocol:
+        description:
+            - protocol for object-group service with port-object
+        choices: ['udp', 'tcp', 'tcp-udp']
+    host_ip:
+        description:
+            - host ip address for object-group network i.e. "network-object host 8.8.8.8"
+        type: list
+    description:
+        description:
+            - object-group description
+    group_object:
+        description:
+            - group-object for network object-group i.e. "network-object object-group ansible_test"
+        type: list
+    ip_mask:
+        description:
+            - IP address and mask for network object-group i.e. "network-object 192.168.0.0 255.255.0.0"
+        type: list
+    port_range:
+        description:
+            - port range for port-object i.e. "port-object range 0 1024"
+    port_eq:
+        description:
+            - single port for port-object i.e. "port-object eq www"
+    service_cfg:
+        description:
+            - service-object configuration protocol/direction/range,port/port
+    state:
+        description:
+            - Manage the state of the resource.
+        default: replace
+        choices: ['present', 'absent', 'replace']
+"""
+
+EXAMPLES = """
+# Note: examples below use the following provider dict to handle
```

Comment:
Please remove this statement. Provider has been deprecated since 2.5. New modules should not have it.